js/details.js

```diff
@@ -0,0 +1,44 @@
+document.addEventListener("DOMContentLoaded", () => {
+  const countryName = new URLSearchParams(window.location.search).get('country');
+  if (countryName) {
+    fetchCountryData(countryName);
+  }
+  else {
+    console.error('Country name not found');
+  }
+});
+
+const fetchCountryData = (countryName) => {
+  fetch('./CountriesData.json')
```

Comment:
Avoid calling fetch repeatedly for the same data (e.g., when filtering regions). Cache the data once: